### ISSUE TITLE
[java] UnnecessaryVarargsArrayCreation should not warn when a method is overridden

### ISSUE THREAD

**Issue by tsmock** [pmd/pmd#5069]
**Affects PMD Version:** 7.2.0

**Rule:** UnnecessaryVarargsArrayCreation

Please provide the rule name and a link to the rule documentation:
https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unnecessaryvarargsarraycreation

**Description:**

**Code Sample demonstrating the issue:**

```java
public class Test {
  public Test() {
    this(new String[] {"prefix"});
  }
  public Test(String otherPrefix) {
    this("prefix", otherPrefix);
  }
  private Test(String... prefixes) {
    for (String str : prefixes) {
      System.out.println(str);
    }
  }
}

```

**Expected outcome:**

PMD reports a violation at line 3, but that's wrong. That's a false positive.
```
[INFO] PMD version: 7.2.0
[INFO] PMD Failure: Test:3 Rule:UnnecessaryVarargsArrayCreation Priority:3 Unnecessary explicit array creation for varargs method call.
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*: Maven
